library-app/src/components/Books/BooksList/EditBookForm.tsx

```diff
@@ -0,0 +1,227 @@
+import axios from 'axios'
+import { FormEvent, useCallback, useEffect, useState } from 'react'
+import Select, { MultiValue } from 'react-select'
+import { Author, AuthorPost } from '../../../models/author.model'
+import { BookBodyDataGet } from '../../../models/bookData.model'
+import { getAuthors, postAuthor } from '../../../services/AuthorServices'
+import { putBookRequest } from '../../../services/BooksServices'
+import styles from './ManageBookForm.module.css'
+
+interface EditBookFormProps {
+  book: BookBodyDataGet
+}
+
+const EditBookForm = ({ book }: EditBookFormProps) => {
+  const [authors, setAuthors] = useState<Author[]>([])
+  const [isOpenForm, setIsOpenForm] = useState(false)
+  const [requestCover, setRequestCover] = useState<Blob>(new Blob())
+  const [cover, setCover] = useState('')
+  const [authorForm, setAuthorForm] = useState<AuthorPost>({
+    FirstName: '',
+    LastName: '',
+  })
+  const [formData, setFormData] = useState<BookBodyDataGet>({
+    Id: book.Id,
+    Title: book.Title,
+    Description: book.Description,
+    Isbn: book.Isbn,
+    Quantity: book.Quantity,
+    Cover: book.Cover,
+    PublishDate: book.PublishDate,
+    Authors: book.Authors,
+  })
+
+  const openFormhandler = () => {
+    setIsOpenForm(!isOpenForm)
+  }
+
+  const handleFileChange = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    if (currentTarget.files) {
+      const files = currentTarget.files
+      const reader = new FileReader()
+      if (files) {
```

Comment:
If you use if (currentTarget.files) to check if that exists  and then assign it to files variable, why do you need additional check if files variable exists if you already know it exists for sure after the first check?